syntax = "proto3";

service BookDetails {
  rpc GetBookDetails (BookDetailsRequest) returns (BookDetailsResponse);
  rpc GetAuthorsList (AuthorsListRequest) returns (AuthorsListResponse);
}

message BookDetailsData {
  string author = 1;
  string title = 2;
  string quotes = 3;
}

message AuthorsListData {
  string author = 1;
}

message BookDetailsRequest {
  string author = 1;
}

message AuthorsListRequest {}

message BookDetailsResponse {
  repeated BookDetailsData bookDetails = 1;
}

message AuthorsListResponse {
  repeated AuthorsListData authorsList = 1;
}
